src/quark-core/src/QuarkNonceManager.sol

```diff
@@ -58,7 +48,8 @@ contract QuarkNonceManager {
             revert InvalidSubmissionToken(msg.sender, nonce, submissionToken);
         }
 
-        nonceSubmissions[msg.sender][nonce] = submissionToken;
+        // Note: even with a valid submission token, we always set non-replayables to exhausted (e.g. for cancelations)
```

Comment:
```suggestion
        // Note: even with a valid submission token, we always set non-replayables to exhausted (e.g. for cancellations)
```

Reply:
changed, though note it's considered a valid (though less common) spelling: https://www.merriam-webster.com/dictionary/cancelation